### PR TITLE
In linting scripts, only lint diff from develop

### LINKED ISSUE #4402: Limit linting and fixing to files changed with develop
## Feature description

Currently running linting commands like `npm run lint:php` is slow because it checks every file. We should limit to just checking the files that have changed in comparison with `develop`. In other words, 

```bash
vendor/bin/phpcs $(git diff develop --name-only | grep php)
```

If no PHP files are changed, then it should also just short-circuit.

This should be done for PHP, JS, and CSS.

This will then allow for the `lint` commands to be added to a `pre-commit` hook.

The `lint:fix` commands could also benefit from this.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by kienstra**
## Summary

* In `package.json` linting scripts, this only lints the diff from `develop`
* ~Adds a pre-commit hook, using [husky](https://www.npmjs.com/package/husky) (maybe out of scope for this PR)~
* ~If we don't all want a pre-commit hook, maybe I should revert it. It's not optional in the way the `wp-dev-lib` pre-commit hook is. It's installed on `npm install`.~
* ~The pre-commit hook [doesn't overwrite](https://www.npmjs.com/package/husky) existing pre-commit hooks~
* ~Please do `npm install` to install the pre-commit hook~

Fixes #4402 

## Checklist

- [x] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/contributing/project-management.md#life-of-an-issue) (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests). (Manually tested, but unit testing doesn't really apply)
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -108,13 +109,13 @@
     "env:stop": "./bin/local-env/stop.sh",
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "wp-scripts lint-style $(git diff develop --diff-filter=d --name-only | grep -E 'css$')",
```
Comment (kienstra) [inline package.json]: `--diff-filter=d` prevents an error from linting a file that was deleted

File: package.json [hunk 2]
```diff
@@ -108,13 +109,13 @@
...
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "wp-scripts lint-js $(git diff develop --diff-filter=d --name-only | grep -E '\\.js$')",
     "lint:js:fix": "npm run lint:js -- --fix",
-    "lint:php": "vendor/bin/phpcs",
+    "lint:php": "vendor/bin/phpcs $(git diff develop --diff-filter=d --name-only | grep -E '\\.php$')",
```
Comment (kienstra) [inline package.json]: Maybe `grep -E '\\.php$'` is overkill as opposed to `grep php`

But `grep -E '\\.js$'` is probably better than simply `grep js`

File: package.json [hunk 3]
```diff
@@ -129,6 +130,11 @@
     "test:php": "phpunit",
     "test:php:help": "npm run test:php -- --help"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run lint"
+    }
+  },
```
Comment (kienstra) [inline package.json]: Maybe this is out of scope for this PR. It [doesn't overwrite](https://www.npmjs.com/package/husky) existing pre-commit hooks. If this is reverted, `package-lock.json` should also be reverted.

File: package.json [hunk 4]
```diff
@@ -129,6 +130,11 @@
     "test:php": "phpunit",
     "test:php:help": "npm run test:php -- --help"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm run lint"
```
Comment (kienstra) [inline package.json]: Maybe this pre-commit hook is out of scope for this PR.

File: package.json [hunk 5]
```diff
@@ -108,13 +109,13 @@
     "env:stop": "./bin/local-env/stop.sh",
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "wp-scripts lint-style $(git diff develop --diff-filter=d --name-only | grep -E 'css$')",
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "wp-scripts lint-js $(git diff develop --diff-filter=d --name-only | grep -E '\\.js$')",
```
Comment (kienstra) [inline package.json]: `--diff-filter=d` prevents an error from linting a file that was deleted

File: package.json (hunk 1 above)
Comment (kienstra) [inline package.json]: `grep -E 'css$'` is to accomodate possible `.scss` files, though I don't think there are any now

**Comment by kienstra** [comment]
Hi Weston,
Hope you had a good day, no hurry with this.

**Comment by kienstra** [comment]
Sorry, not quite ready for review. If there's no diff to develop at all, it looks like `npm run lint:php` lints every PHP file.

File: package.json [hunk 6]
```diff
@@ -108,13 +109,13 @@
...
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "git diff develop --diff-filter=d --name-only | grep -E 'css$' | xargs wp-scripts lint-style",
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "git diff develop --diff-filter=d --name-only | grep -E '\\.js$' | xargs wp-scripts lint-js",
     "lint:js:fix": "npm run lint:js -- --fix",
-    "lint:php": "vendor/bin/phpcs",
+    "lint:php": "git diff develop --diff-filter=d --name-only | grep -E '\\.php$' | xargs vendor/bin/phpcs",
```
Comment (kienstra) [inline package.json]: Maybe `grep -E '\\.php$'` is overkill as opposed to `grep php`

But `grep -E '\\.js$'` is probably better than simply `grep js`

File: package.json [hunk 7]
```diff
@@ -108,13 +109,13 @@
     "env:stop": "./bin/local-env/stop.sh",
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "git diff develop --diff-filter=d --name-only | grep -E 'css$' | xargs wp-scripts lint-style",
```
Comment (kienstra) [inline package.json]: `grep -E 'css$'` is to accomodate possible `.scss` files, though I don't think there are any now

File: package.json [hunk 8]
```diff
@@ -108,13 +109,13 @@
...
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "git diff develop --diff-filter=d --name-only | grep -E '\\.js$' | xargs wp-scripts lint-js",
```
Comment (kienstra) [inline package.json]: `--diff-filter=d` prevents an error from linting a file that was deleted

File: package.json [hunk 9]
```diff
@@ -104,17 +105,18 @@
...
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "npm run diff:develop | grep -E 'css$' | xargs wp-scripts lint-style",
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "npm run diff:develop | grep -E '\\.js$' | xargs wp-scripts lint-js",
     "lint:js:fix": "npm run lint:js -- --fix",
-    "lint:php": "vendor/bin/phpcs",
+    "lint:php": "npm run diff:develop | grep -E '\\.php$' | xargs vendor/bin/phpcs",
```
Comment (kienstra) [inline package.json]: Maybe `grep -E '\\.php$'` is overkill as opposed to `grep php`

But `grep -E '\\.js$'` is probably better than simply `grep js`

File: package.json [hunk 10]
```diff
@@ -104,17 +105,18 @@
...
+    "diff:develop": "git diff develop --diff-filter=d --name-only",
     "env:start": "./bin/local-env/start.sh",
     "env:stop": "./bin/local-env/stop.sh",
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "npm run diff:develop | grep -E 'css$' | xargs wp-scripts lint-style",
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "npm run diff:develop | grep -E '\\.js$' | xargs wp-scripts lint-js",
```
Comment (kienstra) [inline package.json]: `--diff-filter=d` prevents an error from linting a file that was deleted

File: package.json (hunk 9 above)
Comment (kienstra) [inline package.json]: Maybe `grep -E '\\.php$'` is overkill as opposed to `grep php`

But `grep -E '\\.js$'` is probably better than simply `grep js`

File: package.json [hunk 11]
```diff
@@ -104,17 +105,18 @@
     "check-licenses": "wp-scripts check-licenses --production",
     "deploy": "grunt deploy",
     "dev": "wp-scripts start",
+    "diff:develop": "git diff develop --diff-filter=d --name-only",
     "env:start": "./bin/local-env/start.sh",
     "env:stop": "./bin/local-env/stop.sh",
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "npm run diff:develop | grep -E 'css$' | xargs wp-scripts lint-style",
```
Comment (kienstra) [inline package.json]: `grep -E 'css$'` is to accomodate possible `.scss` files, though I don't think there are any now

File: package.json [hunk 12]
```diff
@@ -104,17 +105,18 @@
     "check-licenses": "wp-scripts check-licenses --production",
     "deploy": "grunt deploy",
     "dev": "wp-scripts start",
+    "diff:develop": "git diff develop --diff-filter=d --name-only",
```
Comment (kienstra) [inline package.json]: `--diff-filter=d` prevents an error from linting a file that was deleted

File: package.json [hunk 13]
```diff
@@ -104,17 +105,18 @@
...
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "npm run diff-develop | grep -E 'css$' | xargs wp-scripts lint-style",
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "npm run diff-develop | grep -E '\\.js$' | xargs wp-scripts lint-js",
     "lint:js:fix": "npm run lint:js -- --fix",
-    "lint:php": "vendor/bin/phpcs",
+    "lint:php": "npm run diff-develop | grep -E '\\.php$' | xargs vendor/bin/phpcs",
```
Comment (kienstra) [inline package.json]: Maybe `grep -E '\\.php$'` is overkill as opposed to `grep php`

But `grep -E '\\.js$'` is probably better than simply `grep js`

File: package.json [hunk 14]
```diff
@@ -104,17 +105,18 @@
     "check-licenses": "wp-scripts check-licenses --production",
     "deploy": "grunt deploy",
     "dev": "wp-scripts start",
+    "diff-develop": "git diff develop --diff-filter=d --name-only",
```
Comment (kienstra) [inline package.json]: `--diff-filter=d` prevents an error from linting a file that was deleted

File: package.json [hunk 15]
```diff
@@ -104,17 +105,18 @@
...
     "env:start": "./bin/local-env/start.sh",
     "env:stop": "./bin/local-env/stop.sh",
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "npm run diff-develop | grep -E 'css$' | xargs wp-scripts lint-style",
```
Comment (kienstra) [inline package.json]: `grep -E 'css$'` is to accomodate possible `.scss` files, though I don't think there are any now

**Comment by kienstra** [comment]
This is ready for review, sorry for that earlier.

File: package.json (hunk 4 above)
Comment (westonruter) [reply package.json]: Yes, let's omit that for now.

Comment (kienstra) [reply package.json]: Sure, I'll revert it

Comment (kienstra) [reply package.json]: Removed in https://github.com/ampproject/amp-wp/pull/4441/commits/740f3d130777dc07cda92ec464a762c81eb78890

File: package.json [hunk 16]
```diff
@@ -104,17 +104,18 @@
     "check-licenses": "wp-scripts check-licenses --production",
     "deploy": "grunt deploy",
     "dev": "wp-scripts start",
+    "diff-develop": "git diff develop --diff-filter=d --name-only",
```
Comment (westonruter) [inline package.json]: Perhaps `ls-changed-files` would be more appropriate.

File: package.json [hunk 17]
```diff
@@ -104,17 +104,18 @@
...
     "env:start": "./bin/local-env/start.sh",
     "env:stop": "./bin/local-env/stop.sh",
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "npm run diff-develop | grep -E 'css$' | xargs wp-scripts lint-style",
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "npm run diff-develop | grep -E '\\.js$' | xargs wp-scripts lint-js",
     "lint:js:fix": "npm run lint:js -- --fix",
```
Comment (westonruter) [inline package.json:125]: Missing incorporation of the subset here.

File: package.json [hunk 18]
```diff
@@ -104,17 +104,18 @@
...
     "deploy": "grunt deploy",
     "dev": "wp-scripts start",
+    "diff-develop": "git diff develop --diff-filter=d --name-only",
     "env:start": "./bin/local-env/start.sh",
     "env:stop": "./bin/local-env/stop.sh",
     "env:reset-site": "./bin/local-env/install-wordpress.sh --reset-site",
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "wp-scripts lint-style",
+    "lint:css": "npm run diff-develop | grep -E 'css$' | xargs wp-scripts lint-style",
     "lint:css:fix": "npm run lint:css -- --fix",
```
Comment (westonruter) [inline package.json]: Missing incorporation of the subset.

File: package.json (hunk 15 above)
Comment (westonruter) [reply package.json]:
```suggestion
    "lint:css": "npm run diff-develop | grep -E '\\.s?css$' | xargs wp-scripts lint-style",
```

File: package.json [hunk 19]
```diff
@@ -104,17 +104,18 @@
...
-    "lint:js": "wp-scripts lint-js",
+    "lint:js": "npm run diff-develop | grep -E '\\.js$' | xargs wp-scripts lint-js",
     "lint:js:fix": "npm run lint:js -- --fix",
-    "lint:php": "vendor/bin/phpcs",
+    "lint:php": "npm run diff-develop | grep -E '\\.php$' | xargs vendor/bin/phpcs",
```
Comment (westonruter) [inline package.json]: This isn't working properly when there are no changed PHP files. It is ending up checking _all_ files. Perhaps it is because the `phpcs.xml` has `<file>.</file>`?

File: package.json (hunk 16 above)
Comment (westonruter) [reply package.json]: Alternatively, it may be better to just avoid having a separate command for this, and to instead just replace the instances of `npm run diff-develop` below with just the `git diff` command since it is short, and invoking `git` directly will reduce the need to fire up a subshell. If making this as fast as possible then this would be a consideration.

File: package.json (hunk 19 above)
Comment (kienstra) [reply package.json]: Hm, it looks like that might have been because `npm run diff-develop` also takes account of the changes in the `develop` branch, and this branch didn't have the latest from `develop`.

Before the [commit below](https://github.com/ampproject/amp-wp/pull/4441/commits/7d35852b1a1efa74b1dc71acd1162bc11e0d7edd), here's the result of `npm run diff-develop`

```sh
bin/tag-built.sh
composer.json
composer.lock
contributing/engineering.md
includes/sanitizers/class-amp-base-sanitizer.php
includes/validation/class-amp-validation-error-taxonomy.php
lib/common/src/DevMode.php
lib/common/src/Dom/Document.php
lib/common/tests/DevModeTest.php
lib/common/tests/Dom/DocumentTest.php
package-lock.json
package.json
tests/php/test-amp-dev-mode-sanitizer.php
tests/php/test-class-amp-base-sanitizer.php
tests/php/test-tag-and-attribute-sanitizer.php
tests/php/validation/test-class-amp-validation-error-taxonomy.php
tests/php/validation/test-class-amp-validation-manager.php
```

Comment (kienstra) [reply package.json]: Still, it's not good that it lints files that were changed in `develop`, as those aren't relevant here.

File: package.json (hunk 17 above)
Comment (kienstra) [reply package.json:125]: Sure, is this what you had in mind by 'subset'? https://github.com/ampproject/amp-wp/pull/4441/commits/83e21556ba66a5a33464a9486e883c6ca990c785

File: package.json (hunk 18 above)
Comment (kienstra) [reply package.json]: Sure, is https://github.com/ampproject/amp-wp/pull/4441/commits/83e21556ba66a5a33464a9486e883c6ca990c785 what you had in mind by subset?

File: package.json (hunk 16 above)
Comment (kienstra) [reply package.json]: Sure, applied in https://github.com/ampproject/amp-wp/pull/4441/commits/83e21556ba66a5a33464a9486e883c6ca990c785

File: package.json (hunk 17 above)
Comment (westonruter) [reply package.json:125]: Subset = just the files changed from `develop`

Comment (kienstra) [reply package.json:125]: OK, thanks

**Comment by kienstra** [comment]
Hi @westonruter,
Is your point from https://github.com/ampproject/amp-wp/pull/4441#discussion_r399612591 addressed? 

It's not really ideal, but probably than linting all staged files.

Thanks, Weston!

**Comment by pierlon** [comment]
@googlebot I consent.

**Comment by pierlon** [comment]
This should be a small change that will contribute to the improvement of the DX experience. I've milestoned the PR and linked issue for v2.0.1 with the hopes that it can be reviewed and merged in time.

File: package.json [hunk 20]
```diff
@@ -118,12 +118,12 @@
...
     "lint": "npm-run-all --parallel lint:*",
-    "lint:css": "./bin/get-changed-files.sh '\\.s?css$' | xargs -r wp-scripts lint-style",
+    "lint:css": "git diff HEAD^ --diff-filter=d --name-only | grep -E '\\.s?css$' | xargs -r wp-scripts lint-style",
     "lint:css:fix": "npm run lint:css -- --fix",
-    "lint:js": "./bin/get-changed-files.sh '\\.js$' | xargs -r wp-scripts lint-js",
+    "lint:js": "git diff HEAD^ --diff-filter=d --name-only | grep -E '\\.js$' | xargs -r wp-scripts lint-js",
     "lint:js:fix": "npm run lint:js -- --fix",
-    "lint:php": "./bin/get-changed-files.sh '\\.php$' | xargs -r ./vendor/bin/phpcs",
-    "lint:php:fix": "./bin/get-changed-files.sh '\\.php$' | xargs -r ./bin/phpcbf.sh",
+    "lint:php": "git diff HEAD^ --diff-filter=d --name-only | grep -E '\\.php$' | xargs -r ./vendor/bin/phpcs",
```
Comment (westonruter) [inline package.json]: Why `HEAD^` and not `HEAD`?

Also, invoking via `xargs` is going to mean that we won't be able to take advantage of `<arg name="parallel" value="20"/>` in the PHPCS config.

**Review by westonruter — CHANGES_REQUESTED** [review]

File: package.json (hunk 20 above)
Comment (pierlon) [reply package.json]: > Why `HEAD^` and not `HEAD`?

If I recall correctly it was because of renamed files not being listed when `HEAD` was used.

Nevertheless, it wouldn't be feasible changing these lint scripts here. It would break the workflow of linting the whole project (like in a CI environment, for example).

As discussed a few days ago, I think we can take advantage of a package called [lint-staged](https://github.com/okonet/lint-staged), which gives us fine-grained control on what commands to run when certain files are staged. This would allow us to keep our original lint scripts intact, and with the combination of a pre-commit hook, we could have file changes being automatically linted whenever we're committing, for example.

See https://github.com/ampproject/amp-wp/pull/4441/commits/ab034bce6c1a879c07fedcd4b943ed1413bae14d for a basic setup of what I think is sufficient for the project.

File: package.json [hunk 21]
```diff
@@ -139,6 +141,23 @@
...
   },
+  "lint-staged": {
+    "package.json": [
+      "npm run lint:pkg-json"
+    ],
+    "**/*.css": [
+      "npm run lint:css:fix"
+    ],
+    "**/*.js": [
+      "npm run lint:js:fix"
```
Comment (schlessera) [inline package.json]: Is this "fixing" linting issues here? If so, wouldn't that be incompatible with "staged changes only", as the changes it would make would then not be staged. So you fix something but won't commit that fix, unless you remember to use `git add` again.

**Review by schlessera — DISMISSED** [review]

File: package.json (hunk 21 above)
Comment (pierlon) [reply package.json]: Yes that makes sense. What's more is that when the `--fix` option is specified when linting JS or CSS files, all the eligible files are linted, instead of the staged ones. I'm not sure if it's a linter configuration error or bug with the linter itself.

I've removed the "fixing" of the lint issues in https://github.com/ampproject/amp-wp/pull/4441/commits/76a97167473966a7311c92343fde871436c0434e.

File: package.json [hunk 22]
```diff
@@ -123,9 +124,10 @@
     "lint:js": "wp-scripts lint-js",
     "lint:js:fix": "npm run lint:js -- --fix",
     "lint:php": "vendor/bin/phpcs",
-    "lint:plugin-bootstrap": "vendor/bin/phpcs --runtime-set testVersion 5.2- amp.php",
     "lint:php:fix": "./bin/phpcbf.sh",
+    "lint:plugin-bootstrap": "vendor/bin/phpcs --runtime-set testVersion 5.2- amp.php",
```
Comment (westonruter) [inline package.json:128]: For some reason this command is failing for me, when invoked via `lint-staged`:

```
$ npm run lint:staged

> amp-wp@ lint:staged /app/public/content/plugins/amp
> lint-staged

✔ Preparing...
✔ Hiding unstaged changes to partially staged files...
⚠ Running tasks...
  ↓ No staged files match package.json [SKIPPED]
  ↓ No staged files match **/*.css [SKIPPED]
  ↓ No staged files match **/*.js [SKIPPED]
  ✔ Running tasks for **/!(amp).php
  ❯ Running tasks for amp.php
    ✖ npm run lint:plugin-bootstrap [FAILED]
↓ Skipped because of errors from tasks. [SKIPPED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up...

✖ npm run lint:plugin-bootstrap:
PHP Fatal error:  Uncaught PHP_CodeSniffer\Exceptions\RuntimeException: Undefined index:  in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Files/FileList.php on line 187 in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php:606
Stack trace:
#0 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Files/FileList.php(187): PHP_CodeSniffer\Runner->handleErrors(8, 'Undefined index...', '/app/public/con...', 187, Array)
#1 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php(487): PHP_CodeSniffer\Files\FileList->current()
#2 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php(114): PHP_CodeSniffer\Runner->run()
#3 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/bin/phpcs(18): PHP_CodeSniffer\Runner->runPHPCS()
#4 {main}
  thrown in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php on line 606
PHP Fatal error:  Uncaught PHP_CodeSniffer\Exceptions\RuntimeException: One or more child processes failed to run in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php:544
Stack trace:
#0 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php(114): PHP_CodeSniffer\Runner->run()
#1 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/bin/phpcs(18): PHP_CodeSniffer\Runner->runPHPCS()
#2 {main}
  thrown in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php on line 544
npm ERR! code ELIFECYCLE
npm ERR! errno 255
npm ERR! amp-wp@ lint:plugin-bootstrap: `vendor/bin/phpcs --runtime-set testVersion 5.2- amp.php "/app/public/content/plugins/amp/amp.php"`
npm ERR! Exit status 255
npm ERR!
npm ERR! Failed at the amp-wp@ lint:plugin-bootstrap script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /var/www/.npm/_logs/2020-09-23T03_46_30_988Z-debug.log

> amp-wp@ lint:plugin-bootstrap /app/public/content/plugins/amp
> vendor/bin/phpcs --runtime-set testVersion 5.2- amp.php "/app/public/content/plugins/amp/amp.php"


Fatal error: Uncaught PHP_CodeSniffer\Exceptions\RuntimeException: Undefined index:  in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Files/FileList.php on line 187 in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php:606
Stack trace:
#0 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Files/FileList.php(187): PHP_CodeSniffer\Runner->handleErrors(8, 'Undefined index...', '/app/public/con...', 187, Array)
#1 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php(487): PHP_CodeSniffer\Files\FileList->current()
#2 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php(114): PHP_CodeSniffer\Runner->run()
#3 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/bin/phpcs(18): PHP_CodeSniffer\Runner->runPHPCS()
#4 {main}
  thrown in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php on line 606

Fatal error: Uncaught PHP_CodeSniffer\Exceptions\RuntimeException: One or more child processes failed to run in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php:544
Stack trace:
#0 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php(114): PHP_CodeSniffer\Runner->run()
#1 /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/bin/phpcs(18): PHP_CodeSniffer\Runner->runPHPCS()
#2 {main}
  thrown in /app/public/content/plugins/amp/vendor/squizlabs/php_codesniffer/src/Runner.php on line 544
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! amp-wp@ lint:staged: `lint-staged`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the amp-wp@ lint:staged script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /var/www/.npm/_logs/2020-09-23T03_46_34_383Z-debug.log
```

Comment (westonruter) [reply package.json:128]: Notice how it is invoking `phpcs`:

```bash
vendor/bin/phpcs --runtime-set testVersion 5.2- amp.php "/app/public/content/plugins/amp/amp.php"
```

Notice `amp.php` is being added twice.

File: package.json [hunk 23]
```diff
@@ -139,6 +141,23 @@
...
+      "npm run lint:css"
+    ],
+    "**/*.js": [
+      "npm run lint:js"
+    ],
+    "**/!(amp).php": [
+      "npm run lint:php"
+    ],
+    "amp.php": [
+      "npm run lint:plugin-bootstrap"
```
Comment (westonruter) [inline package.json]: This fixes the issue above when linting `amp.php`:

```suggestion
      "vendor/bin/phpcs --runtime-set testVersion 5.2-"
```

**Review by westonruter — APPROVED** [review]

File: package.json (hunk 22 above)
Comment (pierlon) [reply package.json:128]: Yea good catch. Addressed in https://github.com/ampproject/amp-wp/pull/4441/commits/2c8505746cbedc939ad0dc39821565d6ec96bd7f.
